tools/test.py

```diff
@@ -1581,18 +1593,27 @@ def Main():
     logger.addHandler(fh)
 
   workspace = abspath(join(dirname(sys.argv[0]), '..'))
-  suites = GetSuites(join(workspace, 'test'))
+  test_root = join(workspace, 'test')
+  suites = GetSuites(test_root)
   repositories = [TestRepository(join(workspace, 'test', name)) for name in suites]
   repositories += [TestRepository(a) for a in options.suite]
 
   root = LiteralTestSuite(repositories)
+  default_suites = DefaultSuites(test_root)
```

Comment:
Suggestion: I refactored this a little bit and got:
```python
paths = ArgsToTestPaths(test_root, args, suites)
...

#Instead of DefaultSuites and MaybeSubsystem
def ArgsToTestPaths(test_root, args, suites):
  if len(args) == 0 or 'default' in args:
    return filter(lambda s: s not in IGNORED_SUITES, suites)

  test_dirs = os.listdir(test_root)
  subsystem_regex = re.compile(r'^[a-zA-Z-]*$')
  check = lambda arg: subsystem_regex.match(arg) and (arg not in test_dirs)
  mapped_args = ["*/test*-%s-*" % arg if check(arg) else arg for arg in args]
  paths = [SplitPath(NormalizePath(a)) for a in mapped_args]
  return paths
```

Reply:
Are you thinking we'd pull the IGNORED_SUITES logic into GetSuites()?

Reply:
Ohh I swallowed that...
I think restoring it here is also good.